spec/defines/component/expression_filter_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+require 'yaml'
+
+describe 'rsyslog::component::expression_filter', include_rsyslog: true do
+  let(:title) { 'myexpressionfilter' }
+
+  context 'initial test' do
+    let(:params) do
+      {
+        priority: 55,
+        target: '50_rsyslog.conf',
+        confdir: '/etc/rsyslog.d',
+        conditionals: {
+          if: {
+            expression: 'msg == "test"',
+            tasks: {
+              action: {
+                name: 'myaction',
+                type: 'omfile',
+                config: {
+                  dynaFile: 'remoteSyslog'
+                }
+              }
+            }
+          },
+          else: {
+            tasks: {
+              action: {
+                name: 'myaction2',
+                type: 'omfwd',
+                config: {
+                  KeepAlive: 'on'
+                }
+              }
+            }
+          }
+        }
+      }
+    end
+
+    it do
+      is_expected.to contain_concat__fragment('rsyslog::component::expression_filter::myexpressionfilter').with_content(
+        <<-CONTENT
```

Comment:
I am not a fan of heredoc but everything else looks more awful :(

Reply:
My plan is to remove the content checks in the unit tests when I start working on acceptance tests. Most acceptance tests will be very similar when it comes to testing content and I'd rather test there than here.

Again, mostly consistency right now.